src/test/java/com/pro/sky/employeesaccounting/service/DepartmentServiceTest.java

```diff
@@ -0,0 +1,72 @@
+package com.pro.sky.employeesaccounting.service;
+
+import Model.Employee;
+import exception.EmployeeNotFoundException;
+import org.junit.jupiter.api.Assertions;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.mockito.InjectMocks;
+import org.mockito.Mock;
+import org.mockito.Mockito;
+import service.DepartmentService;
+import service.DepartmentServiceImpl;
+import service.EmployeeService;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Map;
+
+import static org.junit.jupiter.api.Assertions.assertThrows;
+import static org.mockito.Mockito.when;
+import static org.skyscreamer.jsonassert.JSONAssert.assertEquals;
+
+public class DepartmentServiceTest {
+    @InjectMocks
+    private DepartmentService departmentService;
+    @Mock
+    private EmployeeService employeeService;
+
+    private List<Employee> employees = List.of(new Employee("Вячеслав", "Смирнов", 70000, 1),
+            new Employee("Владислав", "Петров", 60000, 2),
+            new Employee("Дмитрий", "Иванов", 71000, 2));
+
+    @Test
+    public void shouldReturnEmployeeWithMaxSalary() {
+        when(employeeService.findAllEmployees()).thenReturn(employees);
+
+        Employee employee = departmentService.findEmployeeMaxSalaryByDepartment(2);
```

Comment:
В твоем случае DepartmentService возвращает Optional, а не Employee. 
```suggestion
        Optional<Employee> employee = departmentService.findEmployeeMaxSalaryByDepartment(2);
```